tuplex/codegen/include/IFailable.h

```diff
@@ -39,7 +59,21 @@ class IFailable {
     void reset() {
         _succeeded = true;
         _messages.clear();
+        _typeError.clear();
     }
+
+    /*!
+     * add all CompileErrors in err to _typeError
+     * @param err
+     */
+    void concatenateTypeError(const std::vector<CompileError> &err) {_typeError.insert(_typeError.begin(), err.begin(), err.end());}
```

Comment:
I think you can leave the Type here away as well. I.e., just use addError(...) addErrors(...), you could even overload +

Reply:
Refactored.